sql-plugin/src/main/spark311/scala/com/nvidia/spark/rapids/shims/DecimalArithmeticOverrides.scala

```diff
@@ -28,6 +28,7 @@
 {"spark": "330cdh"}
 {"spark": "331"}
 {"spark": "332"}
+{"spark": "333"}
```

Comment:
Are these lines actually needed? Every time we add a new snapshot to shim, do we have to add a new line to more than 100 files like this PR? It is too burdensome.

If we actually need these lines (`spark-rapids-shim-json-lines`), can we put them into just one separate file and include it wherever needed?

Reply:
I forgot to mention that the addition of these lines for a new shim was mostly automated, see https://github.com/NVIDIA/spark-rapids/blob/branch-23.04/docs/dev/shimplify.md#adding-a-new-shim which shows how you can start building a new shim based on a closely related shim, and then from there you only need to modify/add the files that are different between the two shims.

Reply:
The rationals for the flexibility this mechanism provides becomes painfully obvious once you go through this exercise for both  upstream and vendor shims.  The design point for the current approach:
- identify shim files for a particular shim with a simple `git grep` without relying on Maven
- makes files self-describing
- maintain the minimum number of source roots that was going out of control because it became the case exceptions  are the rule